identity/src/main/java/com/android/identity/util/ApplicationData.java

```diff
@@ -0,0 +1,163 @@
+     * @param value the value or {@code null} to remove.
+     */
+    void setApplicationData(@NonNull String key, @Nullable Long value);
+
+    /**
+     * Sets application specific data as a {@link Timestamp}.
+     *
+     * <p>Like {@link #setApplicationData(String, byte[])} but encodes the given value
+     * using <a href="http://cbor.io/">CBOR</a> before storing it.
+     *
+     * @param key the key for the data.
+     * @param value the value or {@code null} to remove.
+     */
+    void setApplicationData(@NonNull String key, @Nullable Timestamp value);
+
+    /**
+     * Sets application specific data as a boolean.
+     *
+     * <p>Like {@link #setApplicationData(String, byte[])} but encodes the given value
+     * using <a href="http://cbor.io/">CBOR</a> before storing it.
+     *
+     * @param key the key for the data.
+     * @param value the value or {@code null} to remove.
+     */
+    void setApplicationData(@NonNull String key, @Nullable Boolean value);
+
+    /**
+     * Gets application specific data.
+     *
+     * <p>Gets data previously stored with {@link #getApplicationData(String)}.
+     *
+     * @param key the key for the data.
+     * @return the value or {@code null} if there is no value for the given key.
+     */
+    @Nullable byte[] getApplicationData(@NonNull String key);
+
+    /**
+     * Gets application specific data as a string.
+     *
+     * <p>Like {@link #getApplicationData(String)} but decodes the value as
+     * an UTF-8 string before returning it.
+     *
+     * @param key the key for the data.
+     * @return the value or {@code null} if there is no value for the given key.
+     */
+    @Nullable String getApplicationDataString(@NonNull String key);
+
+    /**
+     * Gets application specific data as a {@link Number}.
+     *
+     * <p>Like {@link #getApplicationData(String)} but decodes the value as
+     * a {@link Long} before returning it.
+     *
+     * @param key the key for the data.
+     * @return the value or {@code null} if there is no value for the given key.
+     */
+    @Nullable Long getApplicationDataNumber(@NonNull String key);
+
+    /**
+     * Gets application specific data as a {@link Timestamp}.
+     *
+     * <p>Like {@link #getApplicationData(String)} but decodes the value as
+     * a {@link Timestamp} before returning it.
+     *
+     * @param key the key for the data.
+     * @return the value or {@code null} if there is no value for the given key.
+     */
+    @Nullable Timestamp getApplicationDataTimestamp(@NonNull String key);
+
+    /**
+     * Gets application specific data as a {@link Boolean}.
+     *
+     * <p>Like {@link #getApplicationData(String)} but decodes the value as
+     * a {@link Boolean} before returning it.
+     *
+     * @param key the key for the data.
+     * @return the value or {@code null} if there is no value for the given key.
+     */
+    @Nullable Boolean getApplicationDataBoolean(@NonNull String key);
```

Comment:
No need to use the boxed type `Boolean`, just use the primitive type `boolean`. Same for setter.